services/venice-controller/src/test/java/com/linkedin/venice/ingestion/control/RealTimeTopicSwitcherTest.java

```diff
@@ -116,4 +121,32 @@ public void testPrepareAndStartReplicationWithNativeReplication() {
     expectedSourceClusters.add(aggregateRealTimeSourceKafkaUrl);
     verify(mockVeniceWriter).broadcastTopicSwitch(eq(expectedSourceClusters), eq(srcTopic), anyLong(), any());
   }
+
+  @Test
+  public void testEnsurePreconditions() {
+    String srcTopic = "testTopic_rt";
+    String destTopic = "testTopic_v1";
+    Store mockStore = mock(Store.class);
+    HybridStoreConfig mockHybridConfig = mock(HybridStoreConfig.class);
+
+    doReturn(true).when(mockStore).isHybrid();
+    doReturn(mockHybridConfig).when(mockStore).getHybridStoreConfig();
+    Version version = new VersionImpl(Version.parseStoreFromKafkaTopicName(destTopic), 1, "test-id");
+    doReturn(Optional.of(version)).when(mockStore).getVersion(Version.parseVersionFromKafkaTopicName(destTopic));
+    doReturn(3600L).when(mockHybridConfig).getRewindTimeInSeconds();
+    doReturn(REWIND_FROM_EOP).when(mockHybridConfig).getBufferReplayPolicy();
+    doReturn(false).when(mockTopicManager).containsTopicAndAllPartitionsAreOnline(srcTopic);
+    doReturn(true).when(mockTopicManager).containsTopicAndAllPartitionsAreOnline(destTopic);
+
+    leaderStorageNodeReplicator.ensurePreconditions(srcTopic, destTopic, mockStore, Optional.of(mockHybridConfig));
+
+    verify(mockTopicManager).createTopic(
```

Comment:
Can we add several different combinations of tests? For example permutation and combination of (VT, RT, VT-ReplicationFactor, RT ReplicationFactor)?

Reply:
Sure, let me add those tests in the appropriate test classes.

Reply:
Actually, the remaining combinations require VeniceHelixAdmin to be unit testable which is not possible right now without major refactoring.